### PR TITLE
betterbib-doi2bibtex: create key based on entry data

### PR DESCRIPTION
**Pull request by tgy**

### CODE REVIEW

**Comment by tgy** [comment]
@nschloe not sure I did it correctly as it's the first time I use git-lfs. I followed the procedure detailed in https://git-lfs.github.com/

**Comment by nschloe** [comment]
Nah, it's still in literal git. You see the entire file in the diff when it really should be just two cryptic lines. (Locally it's always the full txt file of course.)

**Comment by tgy** [comment]
Ok I think it worked this time.

**Comment by nschloe** [comment]
> Ok I think it worked this time.

Yes, nice. Let me go through the code.

File: src/betterbib/cli/doi2bibtex.py [hunk 1]
```diff
@@ -18,20 +21,54 @@ def _get_version_text():
     )
 
 
+def _create_key_for_entry(entry):
+    with open(
+        os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
```
Comment (nschloe) [inline src/betterbib/cli/doi2bibtex.py:26]: Better: `pathlib.Path`.

File: src/betterbib/cli/doi2bibtex.py [hunk 2]
```diff
@@ -18,20 +21,54 @@ def _get_version_text():
     )
 
 
+def _create_key_for_entry(entry):
+    with open(
```
Comment (nschloe) [inline src/betterbib/cli/doi2bibtex.py:25]: Also, we can defer opening the file until `if title`.

File: src/betterbib/cli/doi2bibtex.py [hunk 3]
```diff
@@ -18,20 +21,54 @@ def _get_version_text():
     )
 
 
+def _create_key_for_entry(entry):
```
Comment (nschloe) [inline src/betterbib/cli/doi2bibtex.py:24]: How about adding a test for this? I want to see what this looks like for a bunch of bibtex entries.

File: src/betterbib/cli/doi2bibtex.py [hunk 4]
```diff
@@ -47,4 +84,12 @@ def _get_parser():
         default=sys.stdout,
         help="output file (default: stdout)",
     )
+    parser.add_argument(
```
Comment (nschloe) [inline src/betterbib/cli/doi2bibtex.py:87]: Let's not create an extra cmd option for this. Just make it the default behavior.

**Comment by nschloe** [comment]
I just looked at https://retorque.re/zotero-better-bibtex/citing/. The pattern `[auth:lower][year]` seems simple enough to me?

**Comment by nschloe** [comment]
Could you run `make format`? This makes the linter happy.

**Comment by tgy** [comment]
@nschloe should be good now

File: tests/test_cli.py [hunk 5]
```diff
@@ -1,6 +1,8 @@
 import os
 import tempfile
 
+import pybibtex
```
Comment (nschloe) [inline tests/test_cli.py:4]: What for?

**Comment by nschloe** [comment]
Again, `make format`. The other test are failing, too.

File: tests/test_cli.py (hunk 5 above)
Comment (tgy) [reply tests/test_cli.py]: Initially put the test there, but then moved it to `test_tools`. Leftover import that I forgot to remove. Fixed now

**Comment by tgy** [comment]
`make format` was not running flake8, which causes the issue. One should also run `make lint`

**Comment by nschloe** [comment]
Thanks for the nice contribution!

**Comment by tgy** [comment]
Thanks for being so responsive. Also, I think a "squash and merge" would have led to a cleaner history on the `main` branch because now we end up with all my dirty commit messages, and versions of the code that don't pass the test (it's nice to have each commited new version pass the tests / builds)

**Pull request merged**